packages/sdk/src/evm/core/classes/contract-publisher.ts

```diff
@@ -195,13 +195,13 @@ export class ContractPublisher extends RPCConnectionHandler {
    * @internal
    * TODO clean this up (see method above, too)
    */
-  public async resolveReleasesFromAddress(address: string) {
+  public async resolveContractUriFromAddress(address: string) {
     const contractUri = await resolveContractUriFromAddress(
       address,
       this.getProvider(),
     );
     invariant(contractUri, "Could not resolve contract URI from address");
-    return await this.resolvePublishMetadataFromCompilerMetadata(contractUri);
+    return contractUri;
```

Comment:
can you summarize what the problem was @nachoiacovino ?

Reply:
we weren't passing the provider as before so we were hitting the provider from polygon always instead of hitting the provider from the deployed contract for getting contractUri, so now we're calling this function with the deployed contract chainId then getting releases with the polygon sdk

Reply:
got it that makes sense